src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -406,23 +406,28 @@ async function checkPageExistsAndRedirect(event) {
  * @param {string} url The URL to load version switcher entries from.
  */
 async function fetchVersionSwitcherJSON(url) {
+  const currentPath = getCurrentUrlPath();
   // first check if it's a valid URL
   try {
     var result = new URL(url);
   } catch (err) {
     if (err instanceof TypeError) {
-      if (!window.location.origin) {
-        // window.location.origin is null for local static sites
-        // (ie. window.location.protocol == 'file:')
-        //
-        // TODO: Fix this to return the static version switcher by working out
-        // how to get the correct path to the switcher JSON file on local static builds
-        return null;
-      }
-      // assume we got a relative path, and fix accordingly. But first, we need to
-      // use `fetch()` to follow redirects so we get the correct final base URL
```

Comment:
this comment about using `fetch()` to follow redirects is something I'm now effectively ignoring. @12rambau do you recall the corner case that led to adding this?